File: ComputerNetwork/UDPServer.java
import java.io.IOException;
import java.net.DatagramPacket;
import java.net.DatagramSocket;

public class UDPServer {
    private static final int SERVER_PORT = 9999;
    private static final int BUFFER_SIZE = 1024;

    public static void main(String[] args) {
        try (DatagramSocket serverSocket = new DatagramSocket(SERVER_PORT)) {
            System.out.println("UDP server listening on port " + SERVER_PORT + "...");

            while (true) {
                byte[] receiveData = new byte[BUFFER_SIZE];
                DatagramPacket receivePacket = new DatagramPacket(receiveData, receiveData.length);

                serverSocket.receive(receivePacket);

                String receivedMessage = new String(receivePacket.getData(), 0, receivePacket.getLength());
                System.out.println("Received message: " + receivedMessage);

                String responseMessage = processRequest(receivedMessage);

                byte[] sendData = responseMessage.getBytes();
                DatagramPacket sendPacket = new DatagramPacket(
                    sendData,
                    sendData.length,
                    receivePacket.getAddress(),
                    receivePacket.getPort()
                );

                serverSocket.send(sendPacket);
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    private static String processRequest(String request) {
        String[] parts = request.split(",");
        String action = parts[0];

        switch (action) {
            case "GET_PLACES":
                return getTouristPlaces();
            case "GET_TOURISTS":
                return getTourists();
            case "BOOK_TOUR":
                if (parts.length == 6) {
                    int placeId = Integer.parseInt(parts[1]);
                    int touristId = Integer.parseInt(parts[2]);
                    String guideName = parts[3];
                    double amount = Double.parseDouble(parts[4]);
                    int days = Integer.parseInt(parts[5]);

                    return bookTour(placeId, touristId, guideName, amount, days);
                }
                break;
            default:
                return "Unknown action";
        }

        return "Invalid request";
    }

    private static String getTouristPlaces() {
        // Implement logic to retrieve and return a list of tourist places
        return "Tourist Places:\n1. Beach Resort\n2. Mountain Lodge";
    }

    private static String getTourists() {
        // Implement logic to retrieve and return a list of tourists
        return "Tourists:\n1. Alice\n2. Bob\n3. Charlie";
    }

    private static String bookTour(int placeId, int touristId, String guideName, double amount, int days) {
        // Implement logic to book a tour and return a confirmation message
        return "Booking confirmed:\nPlace ID: " + placeId +
               "\nTourist ID: " + touristId +
               "\nGuide: " + guideName +
               "\nAmount: $" + amount +
               "\nDuration: " + days + " days";
    }
}

File: ComputerNetwork/udpserver.py
import socket

SERVER_IP = "127.0.0.1"
SERVER_PORT = 9999

# Sample data (replace with your data storage or database)
tourist_places = {
    1: "Beach Resort: A beautiful beachside resort.",
    2: "Mountain Lodge: Cozy lodge in the mountains."
}

tourists = {
    1: "Alice",
    2: "Bob",
    3: "Charlie"
}

def main():
    server_socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
    server_socket.bind((SERVER_IP, SERVER_PORT))

    print("UDP server listening on port", SERVER_PORT)

    while True:
        data, client_address = server_socket.recvfrom(1024)
        request = data.decode()
        print("Received request:", request)

        response = process_request(request)
        server_socket.sendto(response.encode(), client_address)

def process_request(request):
    parts = request.split(',')
    action = parts[0]

    if action == "GET_PLACES":
        return get_tourist_places()
    elif action == "GET_TOURISTS":
        return get_tourists()
    elif action == "BOOK_TOUR" and len(parts) == 6:
        place_id, tourist_id, guide_name, amount, days = parts[1:]
        return book_tour(place_id, tourist_id, guide_name, amount, days)
    else:
        return "Invalid request"

def get_tourist_places():
    response = "Tourist Places:"
    for place_id, description in tourist_places.items():
        response += f"\nID: {place_id}, Description: {description}"
    return response

def get_tourists():
    response = "Tourists:"
    for tourist_id, name in tourists.items():
        response += f"\nID: {tourist_id}, Name: {name}"
    return response

def book_tour(place_id, tourist_id, guide_name, amount, days):
    return f"Booking confirmed:\nPlace ID: {place_id}\nTourist ID: {tourist_id}\nGuide: {guide_name}\nAmount: {amount}\nDuration: {days} days"

if __name__ == "__main__":
    main()
